geocoder/location.py

```diff
@@ -121,6 +127,90 @@ def __str__(self):
             return u'{0}, {1}'.format(self.lat, self.lng)
         return u''
 
+
+class BBox(object):
+    """BBox container"""
+
+    DEGREES_TOLERANCE = 0.5
+
+    @classmethod
+    def factory(cls, arg):
+        # validate input first
+        if not isinstance(arg, (list, dict)):
+            raise ValueError(
+                "BBox factory only accept a dict or a list as argument")
+        # we have a dict... just check which fields are given
+        if isinstance(arg, dict):
+            if 'southwest' in arg:
+                return cls(bbox=arg)
+            elif 'bounds' in arg:
+                return cls(bounds=arg['bounds'])
+            elif 'lat' in arg:
+                return cls(lat=arg['lat'], lng=arg['lng'])
+            elif 'west' in arg:
+                return cls(west=arg['west'], south=arg['south'],
+                           east=arg['east'], north=arg['north'])
+            else:
+                raise ValueError(
+                    "Could not found valid values in dict to create a bbox")
+        # we have a list... guess what to call according to the number of parameters given:
+        if len(arg) == 2:
+            lat, lng = arg
+            return cls(lat=lat, lng=lng)
+        elif len(arg) == 4:
+            return cls(bounds=arg)
+        else:
+            raise ValueError(
+                "Could not found valid values in list to create a bbox")
+
+    def __init__(self, bbox=None, bounds=None,
+                 lat=None, lng=None,
+                 west=None, south=None, east=None, north=None):
+        if bbox is not None:
+            self.south, self.west = map(float, bbox['southwest'])
```

Comment:
Just to keep things consistent, we should reverse `bbox` & `bounds` (bbox = 4 number Array).

@ebreton I'll push a commit, but you might need to look over it again to make sure it's good.